### ISSUE TITLE
Unable to resolve "@expo/metro-runtime" from "..."

### ISSUE THREAD

**Issue by Jokerzhzh** [expo/expo#25628]
### Minimal reproducible example

Expo--template

### Which package manager are you using? (Yarn is recommended)

pnpm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

Create a project using `pnpm create-expo-app --template`
Run this with `pnpm start` then an error occurs
![image](https://github.com/expo/expo/assets/102725920/98483840-cfe1-4686-ae94-955f734ceadb)

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 20.9.0 - ~\AppData\Local\fnm_multishells\18256_1701230489013\node.EXE 
      Yarn: 1.22.19 - ~\AppData\Local\fnm_multishells\18256_1701230489013\yarn.CMD
      npm: 10.1.0 - ~\AppData\Local\fnm_multishells\18256_1701230489013\npm.CMD   
    npmPackages:
      expo: ~49.0.15 => 49.0.15 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.6 => 0.72.6 
      react-native-web: ~0.19.6 => 0.19.6 
    Expo Workflow: managed

**Comment by marklawlor** [comment]
I believe, by default, `pnpm` doesn't install peerDependencies so you'll need to add them yourself. Unfortunately its a little bit of a chain of installs (each install had additional peerDependencies). So after you run `pnpm add @expo/metro-runtime` you'll get some other additional missing package errors.

The cause of it is simply because `pnpm` support for React Native is very new (and still unstable), so we're still adjusting the tooling for the change.

@byCedric Pinging you in case there's anything additional here. I'm not 100% across our `pnpm` support

**Comment by Jokerzhzh** [comment]
After running `pnpm add @expo/metro-runtime`, this problem was not solved.

**Comment by marklawlor** [comment]
> After running `pnpm add @expo/metro-runtime`, this problem was not solved.

You may have misunderstood my previous message. Simply adding `@expo/metro-runtime` isn't going to resolve your issue, because `@expo/metro-runtime` also has peerDependanies that need to be install. You will need to repeat the process of installing the missing packages until they server starts. I did quick test and there were approx. 6 missing?

If you find this is too much of a hassle, I would recommend simply using a different package manager. Pnpm support is still unstable

**Comment by byCedric** [comment]
@marklawlor unfortunately, this is not yet solved in SDK 49. See my [other comment about a similar issue](https://github.com/expo/expo/issues/25498#issuecomment-1821389872).

TL;DR; If you are using Expo SDK <=49, you'll still need to set `node-linker=hoisted` inside an **.npmrc** ([docs](https://pnpm.io/npmrc#node-linker)).
![image](https://github.com/expo/expo/assets/1203991/5542d622-4ee4-4ad4-942c-9915b9d32b95)

Besides that, you might still run into some weird dependency issues, which we are aiming to solve starting from SDK 50 (as RN 0.73 also fixed this).

**Comment by ellisio** [comment]
@byCedric Is this resolved in Expo 50? If I remove `node-linker=hoisted` from `.npmrc`, using `pnpm`, I'm getting the following when trying to load our Expo app that lives at `/apps/expo-demo` in a Turborepo.

```
Unable to resolve "@expo/metro-runtime" from "../../node_modules/.pnpm/expo-router@3.4.8_expo-constants@15.4.6_expo@50.0.17_@babel+core@7.24.4_@react-native+babel-p_flq2yhilkqjhpr6judquffiqbi/node_modules/expo-router/entry.js"
```

**Comment by oussaki-dev** [comment]
facing the same issue. Unable to resolve "@expo/metro-runtime" from "node_modules/expo-router/entry.js"

**Comment by rosajsx** [comment]
facing the same issue.

**Comment by caioedut** [comment]
same here

**Comment by zikriacodes** [comment]
after running `npx expo install react-dom react-native-web @expo/metro-runtime` and getting the `_ERESOLVE unable to resolve dependency tree_` error, run `npm i --legacy-peer-deps`

**Comment by zikriacodes** [comment]
add `"react-dom": "^18.2.0", "react-native-web": "~0.19.10", "@expo/metro-runtime": "~3.2.3"` to package.json dependencies
and run `npm i --legacy-peer-deps`

**Comment by vitor3g** [comment]
same here

**Comment by DMyashkov** [comment]
I have the same issue and whatever i do it won't go away!!

**Comment by BIGBOS5** [comment]
Still having the same issue as well
